docs/decorator-reference.md

```diff
@@ -374,6 +375,34 @@ export class User {
 
 Value will be generated only once, before inserting the entity into the database.
 
+#### `@VirtualColumn`
+
+Special column that is never saved to the database and thus acts as a readonly property.
+Each time you call `find` or `findOne` from the entity manager, the value is recalculated based on the query function that was provided in the VirtualColumn Decorator. The alias argument passed to the query references the exact entity alias of the generated query behind the scenes.
+
+```typescript
+@Entity({ name: "companies", alias: "COMP" })
+export class Company extends BaseEntity {
+  @PrimaryColumn("varchar", { length: 50 })
+  name: string;
+
+  @VirtualColumn({ query: (alias) => `SELECT COUNT("name") FROM "employees" WHERE "companyName" = ${alias}.name` })
```

Comment:
I have a question. Can't the `query` be optional?.. we don't need to pass it.. what if we have a big query that returns data from a db with the entity declared column together with the virtual column. How would that work?

Reply:
@CollinCashio

Reply:
what if the virtualcolumn is not in the database at all but some other computed value?

Reply:
background would be graphql since if you want a unique id field and have a composite key for a table you want to cramp that into a single key maybe

Reply:
Hey @Abobos and @sschneider-ihre-pvs ,

This functionality was designed specifically for subqueries. This solves the need of allowing your database engine to return queried result sets and put the data into your defined model (also allows you to use the DB engine to filter data from the model).

If you are just looking for a readonly property that has no need to pull database information and possibly pull from a third party API or some other data source, you could just use a regular property without any column decorator. IE your model could look like:

```ts
@Entity({ name: "timesheets" })
export default class TimeSheet extends BaseEntity {
    @PrimaryGeneratedColumn()
    id: number;

    hours: number;

    @AfterLoad()
    afterLoad() {
         // perform calculations here to set property of hours
    }
}
```

Reply:
@CollinCashio VIrtual Column suffices for my use-case.. but I was thinking you can just add the column name as some alias in a query, without having to do a subquery.

Thanks